mmeval/core/dist_backends/oneflow_dist.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+
+from typing import TYPE_CHECKING, Any, List, TypeVar
+
+from mmeval.utils import try_import
+from .base_backend import BaseDistBackend
+
+if TYPE_CHECKING:
+    import oneflow
+    import oneflow as flow
+    import oneflow.framework.check_point_v2 as check_point_v2
+else:
+    flow = try_import('oneflow')
+    check_point_v2 = try_import('oneflow.framework.check_point_v2')
+
+Tensor = TypeVar('Tensor', bound='oneflow.Tensor')
+
+
+class OneFlowDist(BaseDistBackend):
+    """A distributed communication backend for oneflow."""
+
+    def __init__(self) -> None:
+        super().__init__()
+        if flow is None:
+            raise ImportError(f'For availability of {self.__class__.__name__},'
+                              ' please install oneflow first.')
+
+    @property
+    def is_initialized(self) -> bool:
+        """Returns True if the distributed environment has been initialized.
+
+        Returns:
+            bool: Returns True if the distributed environment has been
+            initialized, otherwise returns False.
+        """
+        try:
+            flow.env.get_world_size()
+            is_init = True
+        except ValueError:
+            is_init = False
+        return is_init
+
+    @property
+    def rank(self) -> int:
+        """Returns the rank index of the current process group."""
+        return flow.env.get_local_rank()
+
+    @property
+    def world_size(self) -> int:
+        """Returns the world size of the current process group."""
+        return flow.env.get_world_size()
+
+    def _all_gather(self, tensor: Tensor) -> List[Tensor]:
+        """All gather the given tensor.
+
+        Args:
+            tensor (Tensor): The tensor for all gather.
+
+        Returns:
+            list: A list of the gathered tensor.
+        """
+        tensor_list = [
+            flow.empty_like(tensor).to(tensor.device)
+            for _ in range(self.world_size)
+        ]
+        flow.comm.all_gather(tensor_list, tensor)
+        return tensor_list
+
+    def _all_gather_py_obj(self, obj: Any) -> List[Any]:
+        """All gather the given python object.
+
+        Args:
+            obj (Any): The python object for all gather.
+
+        Returns:
+            list: A list of the gathered objects.
+        """
+        obj_list = []
+        for src in range(flow.env.get_world_size()):
+            new_obj = check_point_v2._broadcast_py_object(obj, src)
+            obj_list.append(new_obj)
+        return obj_list
```

Comment:
It is recommended to write this as a function `_oneflow_all_gather_object` but not a method of `OneFlowDist`~